packages/idempotency/src/IdempotencyHandler.ts

```diff
@@ -1,25 +1,25 @@
-import type { AnyFunctionWithRecord, IdempotencyOptions } from './types';
+import type { AnyFunctionWithRecord } from './types';
 import { IdempotencyRecordStatus } from './types';
 import {
   IdempotencyAlreadyInProgressError,
   IdempotencyInconsistentStateError,
   IdempotencyItemAlreadyExistsError,
   IdempotencyPersistenceLayerError,
 } from './Exceptions';
-import { IdempotencyRecord } from './persistence';
+import { BasePersistenceLayer, IdempotencyRecord } from './persistence';
+import { IdempotencyConfig } from 'IdempotencyConfig';
 
 export class IdempotencyHandler<U> {
   public constructor(
     private functionToMakeIdempotent: AnyFunctionWithRecord<U>,
     private functionPayloadToBeHashed: Record<string, unknown>,
-    private idempotencyOptions: IdempotencyOptions,
+    private config: IdempotencyConfig,
```

Comment:
A few comments:
- `config` is now unused, is this right? should it be removed?
- Should we make all these arguments an object i.e. `config` like everywhere else? In JS positional arguments are hard to deal with
- Regardless of the above, I think we should change this class declaration to follow the same style as the rest of the project and have the property declared above the constructor, with the constructor actually assigning them as needed

Reply:
* Need to add `persistanceStore.configure(config)` to the constructor
* Do you mean wrapping the signature into an interface `IdempotencyHandlerOptions`?
* agree